spec/main.tex

```diff
@@ -1843,77 +1856,111 @@ \subsection{Provisioning Protocol}
 
 This protocol is run to generate auxiliary information for each signer in a cluster.  
 
+\subsubsection{Utility functions}
+
+\begin{description}
+
+\item
+\begin{inlineAlgorithm}
+\algoName{$\fn{generate\_paillier\_key}^L() \to (N; p, q)$}
+\algoInputs{security level $L = (\kappa_\rsa, \dots)$}
+\algoOutputsList{
+    \item Paillier public key $N \in \Z$ of $n_\rsa$ bits or more
+    \item safe primes $p, q \in \Z^2$, each of $\kappa_\rsa$ bits or more
+}
+\begin{algorithmic}[1]
+\State Generate $\kappa_\rsa$-bit safe primes $p, q$ using the algorithm from Section~\ref{sec:safe-prime}.
+\State Compute Pailier public key $N = pq$
+\State \Return $(N; p, q)$
+\end{algorithmic}
+\end{inlineAlgorithm}
+
+\item
+\begin{inlineAlgorithm}
+\algoName{$\fn{generate\_pedersen\_params}^L() \to (N, s, t; \phi, \lambda)$}
+\algoInputs{security level $L = (\kappa_\rsa, \dots)$}
+\algoOutputs{Pedersen public parameters $N, s, t \in \Z^3$, and secret parameter $\lambda \in \Z$}
```

Comment:
Nitpick: phi not mentioned as output

Reply:
It's private, knowing phi you can easily find factorization. I added it.